logstash-core/src/main/java/org/logstash/common/LsQueueUtils.java

```diff
@@ -45,12 +46,12 @@ public static void addAll(final BlockingQueue<JrubyEventExtLibrary.RubyEvent> qu
      * @throws InterruptedException On Interrupt during {@link BlockingQueue#poll()} or
      * {@link BlockingQueue#drainTo(Collection)}
      */
-    public static Collection<JrubyEventExtLibrary.RubyEvent> drain(
+    public static Set<JrubyEventExtLibrary.RubyEvent> drain(
```

Comment:
I didn't understand this change from `Collection<>` to `Set<>`. So far as I see, this method is called only with a parameter of type `LinkedHashSet`, but I believe we need ordering for certain codecs here, right?

Reply:
@danhermann right this isn't good, this should be `LinkedHashSet` not `Set` my bad, will fix later :)